api/repositories/stack_repository.go

```diff
@@ -0,0 +1,103 @@
+package repositories
+
+import (
+	"context"
+	"fmt"
+	"slices"
+
+	"time"
+
+	"code.cloudfoundry.org/korifi/api/authorization"
+	apierrors "code.cloudfoundry.org/korifi/api/errors"
+
+	korifiv1alpha1 "code.cloudfoundry.org/korifi/controllers/api/v1alpha1"
+	"github.com/BooleanCat/go-functional/v2/it"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+const (
+	StackResourceType = "Stack"
+)
+
+type StackRepository struct {
+	builderName       string
+	userClientFactory authorization.UserK8sClientFactory
+	rootNamespace     string
+}
+
+type StackRecord struct {
+	GUID        string
+	CreatedAt   time.Time
+	UpdatedAt   *time.Time
+	Name        string
+	Description string
+	Labels      map[string]string
+	Annotations map[string]string
+}
+
+func NewStackRepository(
+	builderName string,
+	userClientFactory authorization.UserK8sClientFactory,
+	rootNamespace string,
+) *StackRepository {
+	return &StackRepository{
+		builderName:       builderName,
+		userClientFactory: userClientFactory,
+		rootNamespace:     rootNamespace,
+	}
+}
+
+func (r *StackRepository) GetStacks(ctx context.Context, authInfo authorization.Info) ([]StackRecord, error) {
+	var builderInfo korifiv1alpha1.BuilderInfo
+
+	userClient, err := r.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	err = userClient.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: r.rootNamespace,
+			Name:      r.builderName,
+		},
+		&builderInfo,
+	)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return nil, apierrors.NewResourceNotReadyError(fmt.Errorf("BuilderInfo %q not found in namespace %q", r.builderName, r.rootNamespace))
+		}
+
+		return nil, apierrors.FromK8sError(err, StackResourceType)
+	}
+
+	if !meta.IsStatusConditionTrue(builderInfo.Status.Conditions, korifiv1alpha1.StatusConditionReady) {
+		var conditionNotReadyMessage string
```

Comment:
I would suggest skipping this complex not-ready check. If the builderInfo is not ready, then the error could just say `BuilderInfo %q not ready` without additional details. They do not make sense to cf api users and they would have to contact the operator anyway. kubernetes users can just peak into the builder info status and figure it out themselves

Reply:
Adjusted the error message and removed the complex not-ready check